extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/OpenTelemetryUtil.java

```diff
@@ -48,4 +50,24 @@ public static ContextPropagators mapPropagators(List<String> propagators) {
 
         return ContextPropagators.create(TextMapPropagator.composite(selectedPropagators));
     }
+
+    /**
+     * Converts a list of "key=value" pairs into a map.
+     * Empty entries will be removed.
+     * In case of duplicate keys, the latest takes precedence.
+     *
+     * @param headers nullable list of "key=value" pairs
+     * @return non null map of key-value pairs
+     */
+    public static Map<String, String> convertKeyValueListToMap(List<String> headers) {
```

Comment:
Please review this helper method introduced on `OpenTelemetryUtil` as you suggested.
I didn't find usage of `@Nullable` and `@Nonnull` annotations in the Quarkus codebase so I just documented in the javadoc.